Renaming Issue Templates correctly and adding a pull request template. (#4261)
helm/helm: .github/PULL_REQUEST_TEMPLATE.md

<!--  Thanks for sending a pull request!  Here are some tips for you:

> 1. If you want *faster* PR reviews, read how: https://github.com/helm/helm/blob/master/CONTRIBUTING.md

Reviewer: ```suggestion
1. Make sure to read the Contributing Guide before submitting your PR: https://github.com/helm/helm/blob/master/CONTRIBUTING.md
2. If this PR closes another issue, add 'closes #<issue number>' somewhere in the PR summary. GitHub will automatically close that issue when this PR gets merged. Alternatively, adding 'refs #<issue number>' will not close the issue, but help provide the reviewer more context.
```

Author: done

> 1. If you want *faster* PR reviews, read how: https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md

Reviewer 2: This should link to the Helm contributor guide

Author: Done

-->

**What this PR does / why we need it**:

> **Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

Reviewer: I'd remove this line in favour of above^

**Special notes for your reviewer**:

**If applicable**:

> - [ ] documentation

Reviewer: I'd change these to be a bit more personal:

```suggestion
- [ ] this PR contains documentation
```

Author: done

> - [ ] unit tests

Reviewer: ```suggestion
- [ ] this PR contains unit tests
```

> - [ ] tested backward compatibility (i.e. deploy with previous version, upgrade with this branch)

Reviewer: ```suggestion
- [ ] this PR has been tested for backwards compatibility
```

> **Release note**:

Reviewer 3: If we start using a release note section we should do two things:

1. Add a label to the issue (automatically?) noting the release notes
1. Make sure the issue is filed in a proper milestone

This way someone performing a release can quickly get a list of notes to include.

Reviewer 2: Yeah, we would need to get the automation in place for that

Reviewer: Yeah, let's remove this for now until someone writes up automation for this.

> - [ ] tested backward compatibility (ie. deploy with previous version, upgrade with this branch)

Reviewer 2: Spelling nit: Should be `i.e.`

Author: Done

> **Release note**:

Reviewer 2: I don't think we really leverage the release note, so we may not want to put it here. Let's see what some of the other maintainers here

Author: Ok
